# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/SajidurCodes/flyrank-ml-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

I selected **Lane 2: Refresh / Content Opportunity Scoring**.

The goal of this lane is to identify which content pages should be reviewed first for refresh or improvement based on observable search and engagement signals.

I chose this lane because it focuses on helping reviewers prioritize limited resources. Instead of attempting to automate content decisions, the objective is to build an evidence-based ranking that highlights pages most likely to benefit from human review. This problem is well suited to the starter dataset and provides a clear path toward a practical machine learning solution.

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Research Question

### Decision

Which content pages should be reviewed first?

### Action

Reviewers inspect the highest-ranked pages and decide whether they should be refreshed, expanded, rewritten, monitored, or left unchanged.

### Cost of a Wrong Recommendation

If a low-priority page is ranked too high, reviewers waste valuable time.

If an important declining page is missed, search visibility and user engagement may continue to decrease before action is taken.

### Why Machine Learning?

Simple rule-based systems cannot capture complex interactions among multiple search and engagement signals.

Machine learning can combine these observable signals to prioritize pages more effectively while still providing rankings that support—not replace—human decision making.

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("flyrank-ml-internship-starter/data/raw/content_refresh_anonymized.csv")

df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [17]:
print("Rows:", len(df))
print("Columns:", len(df.columns))
print(df["trend_direction"].value_counts())

Rows: 30000
Columns: 44
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


To understand whether this lane is worth pursuing, I explored the starter dataset before building any model.

Some key observations are:

- The dataset contains **30,000** content records.
- It includes **44** columns describing content characteristics, search performance, and engagement signals.
- The `trend_direction` column shows the following distribution:
  - **Down:** 16,262 pages
  - **Stable:** 5,962 pages
  - **Up:** 4,388 pages
  - **New:** 2,236 pages
  - **Flat:** 1,152 pages

The relatively large number of pages labeled as **down** suggests there are many candidates for content review and prioritization. This makes the dataset suitable for studying content refresh opportunities and building a ranking system that helps reviewers focus on pages most likely to benefit from attention.

These observations support my choice of **Lane 2: Refresh / Content Opportunity Scoring**.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Language

This project identifies pages that may deserve review based on historical search and engagement signals.

The analysis does not prove that refreshing a page will improve its performance.

The results should be interpreted as decision support rather than automated recommendations.

Observed relationships in the data should not be interpreted as causal effects.

In [18]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [*] Every section above is filled — markdown thinking AND the code that backs it
- [*] The notebook runs top to bottom with no errors (Runtime → Run all)
- [*] No client names, URLs, or private queries anywhere
- [*] My claims use careful words: observed, measured, directional, decision-support
- [*] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.